doc/api/n-api.md

```diff
@@ -3395,6 +3396,147 @@ support it:
 
 <!-- it's very convenient to have all the anchors indexed -->
 <!--lint disable no-unused-definitions remark-lint-->
+## Promises
+
+N-API provides facilities for creating Promise objects as described in
+[Section 25.4][] of the ECMA specification. It implements promises as a pair of
+two objects. When you create a promise with `napi_create_promise()` you receive
+a "deferred" object and the promise object itself. The deferred object is a
+JavaScript object tied to the promise object and must be used to resolve the
```

Comment:
@jasnell In my eyes, the subsequent parts of the sentence/paragraph clear that up?